src/lib/parsers/Args.ts

```diff
@@ -298,20 +299,21 @@ export class Args {
 					...options
 				})
 			);
-			if (result === null) break;
-			if (isErr(result)) {
+			if (result.isErr()) {
+				const error = result.unwrapErr();
+				if (error === null) break;
+
 				if (output.length === 0) {
-					// We need to typecast here because TS doesn't resolve that the type from @sapphire/result is identical to that of Lexure
-					return result as Result<ArgType[K][], UserError>;
+					return Result.err(error);
```

Comment:
No need to wrap the error, just return it.
```suggestion
					return result as Result.Err<UserError>;
```